### PR TITLE
Adding helper function for installing multiple php version packages

### PR DESCRIPTION
**Pull request by halkyon**

### CODE REVIEW

File: lib/puppet/parser/functions/ensure_php_packages.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+module Puppet::Parser::Functions
+  newfunction(:ensure_php_packages, :type => :statement) do |arguments|
```
Comment (mateusz) [inline lib/puppet/parser/functions/ensure_php_packages.rb]: Nitpick, but could we document the function and its arguments?

File: lib/puppet/parser/functions/ensure_php_packages.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    if arguments.size > 3 or arguments.size == 0
+      raise(Puppet::ParseError, "ensure_php_packages(): Wrong number of arguments given (#{arguments.size} for 2 or 3)")
+    elsif arguments.size == 3 and !arguments[2].is_a?(Hash)
```
Comment (mateusz) [inline lib/puppet/parser/functions/ensure_php_packages.rb]: We should probably stick with && and || generally, unless the lower precedence is required for something.

File: lib/puppet/parser/functions/ensure_php_packages.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    if arguments[2]
+      defaults = { "ensure" => "present" }.merge(arguments[2])
+    else
+      defaults = { "ensure" => "present" }
+    end
+
+    versions.each do |version|
+      packages.each do |package|
+        # skip mcrypt on 7.2 and greater, the package was removed.
```
Comment (mateusz) [inline lib/puppet/parser/functions/ensure_php_packages.rb]: A bit redundant. Probably just say "Package is now part of core"?

File: spec/functions/ensure_php_packages_spec.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+require "spec_helper"
+
+describe "ensure_php_packages" do
+  it "creates a resource in the catalogue" do
```
Comment (mateusz) [inline spec/functions/ensure_php_packages_spec.rb]: You could add a test case to cover arguments[2]

File: lib/puppet/parser/functions/ensure_php_packages.rb (hunk 3 above)
Comment (halkyon) [reply lib/puppet/parser/functions/ensure_php_packages.rb]: it isn't part of core, it got removed. I'll just ditch the comment

File: lib/puppet/parser/functions/ensure_php_packages.rb (hunk 2 above)
Comment (halkyon) [reply lib/puppet/parser/functions/ensure_php_packages.rb]: fair enough, will do

File: lib/puppet/parser/functions/ensure_php_packages.rb (hunk 1 above)
Comment (halkyon) [reply lib/puppet/parser/functions/ensure_php_packages.rb]: will do!

File: spec/functions/ensure_php_packages_spec.rb (hunk 4 above)
Comment (halkyon) [reply spec/functions/ensure_php_packages_spec.rb]: will do

**Pull request merged**